q.js

```diff
@@ -1502,6 +1506,10 @@ Promise.prototype.keys = function () {
 Q.all = all;
 function all(promises) {
     return when(promises, function (promises) {
+        if (!isArray(arguments[0]) || arguments.length !== 1) {
```

Comment:
I also don't think it's good to prohibit array-likes (like `arguments`). Best bet is checking for a `length` property.